core/shared/src/test/scala/fs2/StreamSuite.scala

```diff
@@ -1031,6 +1031,13 @@ class StreamSuite extends Fs2Suite {
     assertEquals(countChunks(source), 2)
   }
 
+  test("rechunkRandomly sometimes emits everything in a single chunk") {
+    val fiveChunks = Stream(1) ++ Stream(2) ++ Stream(3) ++ Stream(4) ++ Stream(5)
+
+    val source = fiveChunks.rechunkRandomlyWithSeed(0.1, 2.0)(5).chunks.toList
```

Comment:
Also the way `factor` and `nextSize` are implemented now, I don't think `factor` can ever reach `maxFactor`. So for this example `nextSize` can only be 0 or 1.